schemas/kits-schema.json

```diff
@@ -1,148 +1,74 @@
+                "type": "object",
+                "patternProperties": {
+                    "\\w+": {
+                        "type": "string",
+                        "description": "Path to a compiler for this language"
                     }
-                },
-                "cmakeSettings": {
-                    "type": "object",
-                    "patternProperties": {
-                        ".*": {
-                            "description": "Value for the CMake Setting"
-                        }
+                }
+            },
+            "toolchainFile": {
+                "type": "string",
+                "description": "Path to a toolchain file"
+            },
+            "visualStudio": {
+                "type": "string",
+                "description": "Name of the Visual Studio product"
+            },
+            "visualStudioArchitecture": {
+                "type": "string",
+                "description": "Architecture to target"
+            },
+            "environmentVariables": {
+                "type": "object",
+                "patternProperties": {
+                    ".*": {
+                        "type": "string",
+                        "description": "Value for the environment variable"
                     }
-                },
-                "preferredGenerator": {
-                    "type": "object",
-                    "description": "Set a preferred CMake generator for this Kit",
-                    "properties": {
-                        "name": {
-                            "type": "string",
-                            "description": "Name of the generator to use"
-                        },
-                        "platform": {
-                            "type": "string",
-                            "description": "CMake platform for -A argument"
-                        },
-                        "toolset": {
-                            "type": "string",
-                            "description": "CMake toolset for -T argument"
-                        }
-                    },
-                    "required": [
-                        "name"
-                    ]
                 }
-            }
-        },
-        "baseKit": {
-            "allOf": [
-                {
-                    "$ref": "#/definitions/commonOptions"
-                },
-                {
-                    "type": "object",
-                    "properties": {
-                        "name": {
-                            "type": "string",
-                            "description": "Name for this Kit"
-                        }
-                    },
-                    "required": [
-                        "name"
-                    ]
+            },
+            "cmakeSettings": {
+                "type": "object",
+                "patternProperties": {
+                    ".*": {
+                        "description": "Value for the CMake Setting"
```

Comment:
Are this only -D values does this could be a -C or -T?

Reply:
These are only for `-D` options. Passing in a `-C` initializer is a separate topic, but has come up.